php-checks/src/main/java/org/sonar/php/checks/WeakSSLProtocolCheck.java

```diff
@@ -0,0 +1,161 @@
+      "STREAM_CRYPTO_METHOD_SSLv2_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv3_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv23_SERVER",
+      "STREAM_CRYPTO_METHOD_ANY_SERVER",
+      "STREAM_CRYPTO_METHOD_TLS_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_SERVER"
+    )
+  );
+
+  private static final String MESSAGE = "Change this code to use a stronger protocol";
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = CheckUtils.getFunctionName(tree);
+    if (STREAM_CONTEXT_CREATE.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (!arguments.isEmpty()) {
+        checkSSLConfig(arguments.get(0));
+      }
+    }
+    if (STREAM_SOCKET_ENABLE_CRYPTO.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (arguments.size() > 2) {
+        checkWeakProtocol(getAssignedValue(arguments.get(2)), STREAM_SOCKET_ENABLE_CRYPTO);
+      }
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkSSLConfig(ExpressionTree expressionTree) {
+    ExpressionTree param = getAssignedValue(expressionTree);
+    if (!param.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+      return;
+    }
+    ArrayInitializerTree config = (ArrayInitializerTree) param;
+    getProperty(config, "SSL")
+      .flatMap(sslConfig -> {
+        if (sslConfig.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+          return getProperty((ArrayInitializerTree) sslConfig, "crypto_method");
+        }
+        return Optional.empty();
+      })
+      .ifPresent(value -> checkWeakProtocol(value, STREAM_CONTEXT_CREATE));
+  }
+
+  private void checkWeakProtocol(ExpressionTree expressionTree, String functionName) {
+    Stream<ExpressionTree> expressionTrees = expressionTree.is(Tree.Kind.BITWISE_OR)
+      ? getOperands((BinaryExpressionTree) expressionTree)
+      : Stream.of(expressionTree);
+    expressionTrees.forEach(value -> {
+      if (value.is(Tree.Kind.NAMESPACE_NAME)) {
+        NamespaceNameTree cryptoMethod = (NamespaceNameTree) value;
+        List<String> weakProtocols = WEAK_PROTOCOLS.get(functionName);
+        if (weakProtocols != null && weakProtocols.contains(cryptoMethod.name().text())) {
+          context().newIssue(this, cryptoMethod, MESSAGE);
+        }
+      }
+    });
+  }
+
+  private static Stream<ExpressionTree> getOperands(BinaryExpressionTree binaryExpressionTree) {
+    if (binaryExpressionTree.rightOperand().is(Tree.Kind.BITWISE_OR)) {
+      return Stream.concat(
```

Comment:
nitpick: I think this isn't covered, could add a test to cover it

Reply:
Indeed, I forgot to add this test, thanks!